src/test/java/io/jenkins/plugins/opentelemetry/BaseIntegrationTest.java

```diff
@@ -75,33 +77,34 @@ public class BaseIntegrationTest {
         GitSCM.ALLOW_LOCAL_CHECKOUT = true;
     }
 
-    public static final AtomicInteger jobNameSuffix = new AtomicInteger();
+    protected static final AtomicInteger jobNameSuffix = new AtomicInteger();
 
-    @ClassRule
-    public static BuildWatcher buildWatcher = new BuildWatcher();
+    @SuppressWarnings("unused")
+    @RegisterExtension
+    private static final BuildWatcherExtension BUILD_WATCHER = new BuildWatcherExtension();
 
-    @ClassRule
-    @ConfiguredWithCode("jcasc-elastic-backend.yml")
-    public static JenkinsConfiguredWithCodeRule jenkinsRule = new JenkinsConfiguredWithCodeRule();
+    protected static JenkinsRule jenkinsRule;
 
-    @Rule
-    public ExtendedGitSampleRepoRule sampleRepo = new ExtendedGitSampleRepoRule();
+    @RegisterExtension
+    protected final ExtendedGitSampleRepoRule sampleRepo = new ExtendedGitSampleRepoRule();
 
     static JenkinsControllerOpenTelemetry jenkinsControllerOpenTelemetry;
 
-    @Before
-    public void before() throws Exception {}
+    @BeforeEach
+    void beforeEach() {}
 
-    @After
-    public void after() throws Exception {
+    @AfterEach
+    void afterEach() throws Exception {
         jenkinsRule.waitUntilNoActivity();
         InMemoryMetricExporterProvider.LAST_CREATED_INSTANCE.reset();
         InMemorySpanExporterProvider.LAST_CREATED_INSTANCE.reset();
     }
 
-    @BeforeClass
-    public static void beforeClass() throws Exception {
+    @BeforeAll
+    static void beforeAll(JenkinsRule rule) throws Exception {
         LOGGER.log(Level.INFO, "beforeClass()");
+        jenkinsRule = rule;
+        ConfigurationAsCode.get().configure(BaseIntegrationTest.class.getResource("jcasc-elastic-backend.yml").toExternalForm());
```

Comment:
Are you sure it is not possible to use the annotation `@ConfiguredWithCode("jcasc-elastic-backend.yml")` in the tests or other place? I have used it in other JUnit 5 tests.

Reply:
It is only possible to define `ConfiguredWithCode` on test methods, but not on lifecylce methods (`@BeforeEach` and similar) - however that is possible with `JenkinsRule`. Not using a lifecycle method makes it impossible for inheritors to access the member variable for the rule.

My approach here produces the same result as `ConfiguredWithCode` would with no real downside I could think of.